cmake/Platform/System/LinuxDistDetector.cmake

```diff
@@ -6,6 +6,11 @@ function(detect_host_linux_distribution)
         message(AUTHOR_WARNING "Linux distribution detection called on Apple platform")
     else () # Linux host
         find_program(lsb_release_exec lsb_release)
+
+        if ("lsb_release_exec-NOTFOUND" STREQUAL "${lsb_release_exec}")
+            message(WARNING "Linux distribution couldn't be detected because `lsb_release' not found.")
```

Comment:
There's a mismatch in the ticks here: `lsb_release' i.e The first is a back-tick, 2nd is a single-quote/tick.
I'd choose single-quotes here, so it'd become 'lsb_release'.

Also, I myself choose restrain from the word 'because' in technical information messages. IMHO a hyphen (`-`) is a much better choice, but I'll leave that for your judgement.

At last, I feel that saying a certain "thing" hasn't been found is not informative enough.
Correct me if I'm wrong, but I think that most of the users don't know that the **lsb_release** program is, so they'll just end up opening an issue here, asking for help. Instead, we could inform them that it's a package they need to install. 
So I would rephrase it as such:
> Linux distribution couldn't be detected - Please install 'lsb_release'.

Reply:
I agree with your suggestions!